CMakeLists.txt

```diff
@@ -27,7 +27,7 @@ add_subdirectory(src)
 add_subdirectory(test)
 
 if(MSVC)
-  target_compile_options(LightCombatManager PRIVATE /W4 /WX)
+  target_compile_options(LightCombatManager PRIVATE /W3)
```

Comment:
I'd suggest not reducing the warning level. Instead we should aim on fixing all thrown compiler warnings.